### PR TITLE
feat(query): add support for rsubset in :semver selectors

### PR DESCRIPTION
**Pull request by lukekarrys**

### CODE REVIEW

**Comment by nlf** [comment]
should probably add it to the docs too?

**Comment by lukekarrys** [comment]
Added the function name and a note about what it does to `dependency-selectors.md` docs page.

File: workspaces/arborist/test/query-selector-all.js [hunk 1]
```diff
@@ -712,6 +712,7 @@ t.test('query-selector-all', async t => {
     [':semver(^1.1.1, [version], lte)', []],
     [':semver(^14.0.0, :attr(engines, [node]), intersects)', ['bar@2.0.0']],
     [':semver(>=14, :attr(engines, [node]), subset)', ['abbrev@1.1.1', 'bar@2.0.0']],
+    [':semver(>=14, :attr(engines, [node]), rsubset)', ['bar@2.0.0']],
```
Comment (wraithgar) [inline workspaces/arborist/test/query-selector-all.js:715]: naive question but is this functionally different than `':semver(:attr(engines, [node]), >=14, subset)'`

**Comment by lukekarrys** [comment]
Converting to draft while we work through questions about reversing the order of arguments instead of using the `r` prefix on functions.

**Pull request closed without merging**